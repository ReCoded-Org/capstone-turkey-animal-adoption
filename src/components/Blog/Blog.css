.articleContianer {
  text-align: left;
}

.blogContianer {
  text-align: left;
  margin-bottom: 2rem;
}

.author {
  display: flex;
  align-items: center;
}

.authorImg {
  max-width: 80px;
  height: 80px;
  border-radius: 100%;
  margin-right: 2rem;
}

/* for override purposes */
#articleTitle {
  margin-top: 2rem;
  color: #ff5555;
}

#publishedDate {
  color: #96e8f0;
  font-weight: bold;
}

#blogCol {
  padding: 0rem 7rem;
}

#authorName {
  color: #ff5555;
  font-weight: bold;
}

@media screen and (max-width: 767px) {
  .blog {
    margin-top: 2rem;
  }
  #blogCol {
    padding: 3rem 1rem;
  }
}
