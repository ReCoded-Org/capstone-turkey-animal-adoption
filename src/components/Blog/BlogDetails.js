// import React from "react";
// import { Container, Row, Col } from "react-bootstrap";

// const BlogDetails = ({ blogContent }) => {
//     console.log(Object.values(blogContent))
//     return (
//         <>
//             {Object.values(blogContent).map((blog) => (
//                 <div>
//                     <p>{blog.publishedDate}</p>
//                     <p>{blog.blogTitle}</p>
//                     <p>{blog.author}</p>
//                     <p>{blog.authorWork}</p>
//                 </div>
//             ))}
//         </>
//     )
// }

// export default BlogDetails