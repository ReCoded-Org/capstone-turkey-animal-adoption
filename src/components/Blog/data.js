export let blogContent = {
      firstArticle : {
        id: 1,
        img: "https://i.redd.it/l5tybjlw58w41.jpg",
        blogTitle: "Average Cat Weight – How to Tell if Your Cat is Overweight",
        author: "Bertram Jensen",
        publishedDate: "21 HOURS AGO",
        blogDescription: "Is there such a thing as average cat weight? According to Purina’s experts, an average weight or range will vary from breed to breed and even between males and females. Learn why here, along with what they suggest using instead.....",
        authorWork: "Author at Panoply Store",
        authorImg: "https://images1.penguinrandomhouse.com/author/223095"
      },
      secondArticle : {
        id: 2,
        img: "https://i.redd.it/l5tybjlw58w41.jpg",
        blogTitle: "Four new breeds have been recognized by the American Kennel Club (AKC)",  
        author: "Bertram Jensen",
        publishedDate: "21 HOURS AGO",
        blogDescription: "Is there such a thing as average dog weight? According to Purina’s experts, an average weight or range will vary from breed to breed and even between males and females. Learn why here, along with what they suggest using instead.....",
        authorWork: "Author at Panoply Store",
        authorImg: "https://images1.penguinrandomhouse.com/author/223095"
      },
      thirdArticle : {
        id: 3,
        img: "https://i.redd.it/l5tybjlw58w41.jpg",
        blogTitle: "Four new breeds have been recognized by the American Kennel Club (AKC)",
        author: "Bertram Jensen",
        publishedDate: "21 HOURS AGO",
        blogDescription: "Is there such a thing as average dog weight? According to Purina’s experts, an average weight or range will vary from breed to breed and even between males and females. Learn why here, along with what they suggest using instead.....",
        authorWork: "Author at Panoply Store",
        authorImg: "https://images1.penguinrandomhouse.com/author/223095"
      }
  }