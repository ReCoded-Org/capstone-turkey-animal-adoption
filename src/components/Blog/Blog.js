import React, { useState, useEffect } from "react";
import BlogDetails from "./BlogDetails";
import { Container, Row, Col, Spinner } from "react-bootstrap";

const Blog = () => {
  const [allPosts, setAllPosts] = useState([]);

  const getAllPosts = async () => {
    const data = await fetch(
      "https://public-api.wordpress.com/wp/v2/sites/ecommerce458750409.wordpress.com/posts"
    );

    const posts = await data.json();
    setAllPosts(posts);
  };

  useEffect(() => {
    getAllPosts();
  }, []);

  return allPosts.length > 0 ? (
    <BlogDetails blogContent={allPosts} />
  ) : (
    <Container>
      <Row>
        <Col>
          <Spinner animation="grow" />
        </Col>
      </Row>
    </Container>
  );
};

export default Blog;
