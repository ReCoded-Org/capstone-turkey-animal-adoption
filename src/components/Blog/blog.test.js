import React from "react";
import renderer from "react-test-renderer";
import Blog from "./Blog.js";

it("renders Blog correctly", () => {
  const tree = renderer.create(<Blog />).toJSON();
  expect(tree).toMatchSnapshot();
});
