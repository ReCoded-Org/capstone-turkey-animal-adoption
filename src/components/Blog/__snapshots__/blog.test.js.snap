// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Blog correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="spinner-grow"
      />
    </div>
  </div>
</div>
`;
